src/canvas/SelectableCanvas.ts

```diff
@@ -860,15 +860,27 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
     pointer: Point
   ): FabricObject | undefined {
     const target = this._searchPossibleTargets(objects, pointer);
-    // if we found something in this.targets, and the group is interactive, return that subTarget
+
+    // if we found something in this.targets, and the group is interactive, return the innermost subTarget
+    // that is still interactive
     // TODO: reverify why interactive. the target should be returned always, but selected only
     // if interactive.
-    return target &&
+    if (
+      target &&
       isCollection(target) &&
       target.interactive &&
       this.targets[0]
-      ? this.targets[0]
-      : target;
+    ) {
+      /** {@link this.targets} is ordered bottom-up */
+      let subTarget = this.targets[0];
+      while (subTarget.parent && !subTarget.parent.interactive) {
+        subTarget = subTarget.parent;
+      }
+
+      return subTarget;
```

Comment:
```suggestion
      /** targets[0] is the innermost nested target, but it could be inside non interactive groups and so not a selection target */
      const targets = this.targets;
      for (let i = targets.length - 1; i > 0; i--) {
        if (!targets[i].interactive) {
          // one of the subtargets was not interactive. that is the last subtarget we can return.
          // we can't dig more deep;
          return targets[i];
        }
      }
      return targets[0];
```

This is my version of this problem @jiayihu  @ShaMan123